# HG changeset patch
# User Roland Haas <roland.haas@physics.gatech.edu>
# Date 1315525123 14400
# Node ID 9b019981aa2f56a1bfac986ce71ffb7502e50b70
# Parent  7f69434460672283638d6e024fa7e97d8ee7afcd
Carpet: include unistd to make prototype of getpid visible
also include explicit cast to int from pid_t

diff --git a/Carpet/Carpet/src/SetupGH.cc b/Carpet/Carpet/src/SetupGH.cc
--- a/Carpet/Carpet/src/SetupGH.cc
+++ b/Carpet/Carpet/src/SetupGH.cc
@@ -30,6 +30,7 @@
 #ifdef HAVE_SCHED_H
 #  include <sched.h>
 #endif
+#include <unistd.h>
 
 #include <carpet.hh>
 #include "Timers.hh"
@@ -252,7 +253,7 @@
       MPI_Get_processor_name (hostnamebuf, &hostnamelen);
       string const hostname (hostnamebuf);
 #endif
-      int const mypid = getpid ();
+      int const mypid = static_cast<int> (getpid ());
       // Output
       CCTK_VInfo (CCTK_THORNSTRING,
                   "This process runs on host %s, pid=%d",
